From 084f9d1748066b6dea8164c6d1dac3b8ac153d8e Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Wed, 22 Nov 2017 18:29:31 +0800
Subject: [PATCH 2874/5242] MLK-16941: hdmi: check video mode with cmdline
 mode

commit  ba9e6e058f498776f5be2958e91d7f2ccfc4a4a0 from
https://source.codeaurora.org/external/imx/linux-imx.git

when EDID function is disabled, the cmdline mode is
the max support video mode.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
(cherry picked from commit 4bdf485762441578a50a1b1586777f1291dd32e5)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index 885334c..f245606 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -716,6 +716,14 @@ static int imx_hdp_connector_get_modes(struct drm_connector *connector)
 	struct imx_hdp *hdp = container_of(connector, struct imx_hdp,
 					     connector);
 	enum drm_mode_status mode_status = MODE_OK;
+	struct drm_cmdline_mode *cmdline_mode;
+	cmdline_mode = &connector->cmdline_mode;
+
+	/* cmdline mode is the max support video mode when edid disabled */
+	if (!hdp->is_edid)
+		if (cmdline_mode->xres != 0 &&
+			cmdline_mode->xres < mode->hdisplay)
+			return MODE_BAD_HVALUE;
 
 	if (hdp->is_4kp60 && mode->clock > 594000)
 		return MODE_CLOCK_HIGH;
-- 
1.7.9.5

